elements/pl-order-blocks/pl-order-blocks.py

```diff
@@ -0,0 +1,390 @@
+import prairielearn as pl
+import lxml.html
+import random
+import chevron
+import base64
+import os
+import json
+import re
+
+# Read https://prairielearn.readthedocs.io/en/latest/devElements/
+# Official documentation on making custom PL
+
+
+def render_html_colour(score):
+    # used to render the correct colour depending on student score
+    if score == 0:
+        return 'badge-danger'
+    elif score == 1.0:
+        return 'badge-success'
+    else:
+        return 'badge-warning'
+
+# courtesy of https://github.com/PrairieLearn/PrairieLearn/pull/2625
+
+
+def get_all_answer(submitted_blocks, block_indents, leading_code, trailing_code):
+    if len(submitted_blocks) == 0:
+        return ''
+    answer = ''
+    for index, answer in enumerate(submitted_blocks):
+        indent = int(block_indents[index])
+        answer += ('    ' * indent) + submitted_blocks[index] + '\n'
+    answer = leading_code + '\n' + answer + trailing_code + '\n'
+    return answer
+
+
+def get_trail_answer(submitted_blocks, block_indents, trailing_code):
+    if len(submitted_blocks) == 0:
+        return ''
+    answer = ''
+    for index, answer in enumerate(submitted_blocks):
+        indent = int(block_indents[index])
+        answer += ('    ' * indent) + submitted_blocks[index] + '\n'
+    answer = answer + trailing_code + '\n'
+    return answer
+
+
+def get_lead_answer(submitted_blocks, block_indents, leading_code):
+    if len(submitted_blocks) == 0:
+        return ''
+    answer = ''
+    for index, answer_indent in enumerate(block_indents):
+        indent = int(answer_indent)
+        answer += ('    ' * indent) + submitted_blocks[index] + '\n'
+    answer = leading_code + '\n' + answer
+    return answer
```

Comment:
There's still a lot of duplicated code between these functions that should be abstracted.

Reply:
fixed in latest commit